docs/resources/library.md

```diff
@@ -0,0 +1,115 @@
+---
+subcategory: "Compute"
+---
+# databricks_library resource
+
+Installs a library on [databricks_cluster](cluster.md). Each different type of library has a slightly different syntax. It's possible to set only one type of library within one resource. Otherwise, the plan will fail with an error. 
+
+-> **Note** `databricks_library` resource would always start the associated cluster if it's not running, so make sure to have auto-termination configured. It's not possible to atomically change the version of the same library without cluster restart.
+
+## Installing library on all clusters
+
+You can install libraries on all clusters with the help of [databricks_clusters](../data-sources/clusters.md) data resource:
+
+```hcl
+data "databricks_clusters" "all" {
+}
+
+resource "databricks_library" "cli" {
+  for_each = data.databricks_clusters.all.ids
+  cluster_id = each.key
+  pypi {
+    package = "databricks-cli"
+  }
+}
+```
+
+## Java/Scala JAR
+
+Installing JAR artifacts on a cluster. Location can be anything, that is DBFS or mounted object store (s3, adls, ...)
+
+```hcl
+resource "databricks_dbfs_file" "app" {
+  source = "${path.module}/app-0.0.1.jar"
+  path = "/FileStore/app-0.0.1.jar"
+}
+
+resource "databricks_library" "app" {
+  cluster_id = databricks_cluster.this.id
+  jar = databricks_dbfs_file.app.dbfs_path
+}
+```
+
+## Java/Scala Maven
+
+Installing artifacts from Maven repository. You can also optionally specify a `repo` parameter for custom Maven-style repository, that should be accessible without any authentication for the network that cluster runs in. It can even be properly configured [maven s3 wagon](https://github.com/seahen/maven-s3-wagon), [AWS CodeArtifact](https://aws.amazon.com/codeartifact/) or [Azure Artifacts](https://azure.microsoft.com/en-us/services/devops/artifacts/).
```

Comment:
Not sure about "without any authentication for the network that cluster runs in" -> Maven libraries are resolved in the control plane, so repo should be accessible from it